### ISSUE TITLE
Feature request: Import/export spaces/collections/request as file

### ISSUE THREAD

**Issue by ya-makariy** [httpie/desktop#22]
# Checklist
 
- [x] I've searched for similar feature requests.

# Enhancement request
To add ability to import/export spaces/collections/request as file

# Problem it solves
A convenient way to exchange requests between users

**Comment by mrcsxsiq** [comment]
I came here to suggest this too

**Comment by claudiatd** [comment]
We are working on it @ya-makariy, @mrcsxsiq and @fers4t :) I'll keep you posted!

**Comment by jazuly** [comment]
I came here to suggest this too, really need import/export to sharing space..

**Comment by sounden** [comment]
Any updates on this? would be wonderful to export for backup/migrate. Love the desktop app!

**Comment by cjol** [comment]
A related use-case (depending on the file format) would allow versioning of requests through git, so that requests can be synced alongside the code changes in development.

**Comment by bryan-lott** [comment]
Any progress on this? I'm switching laptops today and would love to be able to export/import.

It appears that the desktop app doesn't store settings in `~/.config` so I'm unable to use @bahua's solution.

**Comment by gtakat** [comment]
I was looking for this method because I replaced the PC.
(from an Intel Mac to an M2 Mac)

I don't think it's a very good method, but I was able to migrate by copying the entire directory below.

- `/Users/{userName}/Library/Application Support/HTTPie`

I adjusted the destination version with the original version.
It's working fine for now.

It is not a recommended method, so please be careful if you imitate it.

**Comment by claytonjroberts** [comment]
Bump. This is important for collaboration, source control, etc.

**Comment by Nahuel92** [comment]
Please bump, I really need this to share collections with my team.

**Comment by UBy** [comment]
Just ran into this issue. 
Since there was an import option, I assumed export would be available as well. So surprised when I didn't find it.
Have to switch to another tool until this feature gets added.
The directory copy is just too much work, if you share regularly.

**Comment by heli-os** [comment]
I checked the discord for something related to this request. so, have a roadmap of upcoming releases that you're anticipating?

![Screenshot 2023-10-26 at 12 29 17](https://github.com/httpie/desktop/assets/44586666/9c6f0d66-3984-4064-aa40-227979ec6531)

**Comment by niqdev** [comment]
In addition to the collection export to share it with the team, I'd like also to be able to export a single request `as cURL`, the same way I can do an import!

**Comment by rnd-debug** [comment]
Would be interested in seeing an "export as OpenAPI spec" or "export as cURL" feature implemented (or help to implement it).

**Comment by terminal-root** [comment]
After trying httpie and seeing the 1.5 year sitting time for this issue, I have gathered local usage is not a priority, which is okay -- make your project how you like. Though, if you are looking for a local-first/privacy-focused option, I recently found [Bruno](https://www.usebruno.com/). This has been a great Postman replacement for my needs (multi-environment request collections and easy pre/post-request scripting).

**Comment by Nahuel92** [comment]
> After trying httpie and seeing the 1.5 year sitting time for this issue, I have gathered local usage is not a priority, which is okay -- make your project how you like. Though, if you are looking for a local-first/privacy-focused option, I recently found [Bruno](https://www.usebruno.com/). This has been a great Postman replacement for my needs (multi-environment request collections and easy pre/post-request scripting).

Thank you so much, that is exactly what I have been looking for since Postman forced people to either use their lightweight client or pay their cloud service.

I'll switch over to Bruno and will encourage my team to do the same.

**Comment by sokol8** [comment]
guys, really really need a way to share several dozens of requests with my colleagues. Is it so complicated to make an export/import feature while you are already support import of Postman collection? Pleeeease

**Comment by EF09FE** [comment]
It's been two years and you still don't support export, this is really disappointing to me, I had to delete httpie and use postman instead

**Comment by sokol8** [comment]
yeah. I have to move and try out Bruno instead (I hate Postman's bloated UI).
I cannot sit stuck on my API collections when I cannot share all that knowledge with my colleagues.

**Comment by gpellizzoni** [comment]
Real bummer that such a vital feature for team collab is missing.

**Comment by KRC2000** [comment]
I kinda assumed that this feature is in place =(

**Comment by DonChunior** [comment]
I would be very pleased if this suggestion for improvement were implemented! :pray:

**Comment by rmdes** [comment]
I'm about to switch to Bruno or return to Postman because this essential feature, which should not even be part of a future team premium is not available by design. 

Put the sync cloud into premium and instead give everyone the export feature, that would actually be a solution for everyone

**Comment by HassanJbara** [comment]
> I'm about to switch to Bruno or return to Postman because this essential feature, which should not even be part of a future team premium is not available by design.
> 
> Put the sync cloud into premium and instead give everyone the export feature, that would actually be a solution for everyone
> 
> edit: sorry, but I'm really frustrated after using it for over a year and only finding now that there is no export, I feel in jail cell, stuck with countless collections and requests that will need to be moved by hand. This is a nightmare.

ditto. This is unacceptable.

**Comment by usswiss** [comment]
So a bullshit!
I think, they don't want spend us this feature.
I had install HTTTPie yesterday first, and play a little bit, and my first question and solution searching point after create some requests was "who can I save it and share with my other PCs and VMs Installations." It seems to be NOT POSSIBLE!
Only with a fucking cloud way ? Forget it, I will deinstall this tool now. It's not to use how I want.

**Comment by IvanLi-CN** [comment]
It's so frustrating.

I've been looking around in the interface for a while today and I really didn't realize there was no export function. I was trying to back up my data, but now I have to hurry up and manually move it to another platform while it's not too much. 💢

**Comment by danieleteti** [comment]
> We are working on it @ya-makariy, @mrcsxsiq and @fers4t :) I'll keep you posted!

How do the work going?

**Comment by lenzch** [comment]
I ran into the same trap as well. Right now [Bruno](https://github.com/usebruno/bruno) looks like a viable alternative.

**Comment by HonzaKubita** [comment]
This is disappointing I really liked httpie but I need to share bunch of requests with my team and it doesn't look like this feature is ever gonna get implemented. So I'll have to make a switch back to insomnia ig.

**Comment by feketegy** [comment]
> This is disappointing I really liked httpie but I need to share bunch of requests with my team and it doesn't look like this feature is ever gonna get implemented. So I'll have to make a switch back to insomnia ig.


Spaces are their premium offering. I don't think this feature will be implemented ever.

**Comment by lannotimo** [comment]
Just came over from Bruno, now I find out that I cannot export spaces... guys... what is this.

**Comment by mikaelvesavuori** [comment]
+1. This is a critical feature. Love httpie and the work done here, but this seems like a fairly standard feature to have in place, especially after all these years?

**Comment by bahua** [comment]
No code updates to the project in 2 years. Is this abandoned?

**Comment by Ashaxer** [comment]
New version released and still this critical feature is not added. it's been almost 3 years...
I really like HTTPie but absence of this basic feature bothers me.

**Comment by druellan** [comment]
A new version is out and this feature is still missing, this is starting to bother me, guys, I don't want to think this is a retention strategy. You have a great app here, people are not going to migrate overnight, just give us the choice.

**Comment by feketegy** [comment]
They have this feature in their paid app, this feature never going to land in the free version.

**Comment by yusufmalikul** [comment]
> They have this feature in their paid app, this feature never going to land in the free version.

where is the link to the paid app? @feketegy 
I can't find it. The desktop version said it's public beta: https://httpie.io/desktop

**Comment by Inffinite** [comment]
> They have this feature in their paid app, this feature never going to land in the free version.

I figured. That said i am not paying to be able to export my collections. Moving on to other apps.

**Comment by jkbrzt** [comment]
**✅ Released in HTTPie Desktop 2025.2.0** 

Export/import spaces, environments, collections, and individual requests as JSON:

<img width="745" alt="Image" src="https://github.com/user-attachments/assets/22fe202b-1ce3-41a2-ac0f-3cf17cd42975" />

<br>
<br>

* Full changelog: https://httpie.io/blog/changelog-0027
* Documentation: https://httpie.io/r/help/export-from-httpie

**Comment by sokol8** [comment]
@jkbrzt finally! Thank you soooo much :) 
my laziness to move out from otherwise wonderful httpie seems to finally pay off ;)

**Comment by Misoramenn** [comment]
Incredible timing. I use HTTPie since one week, today the first time i was in need of an export feature. stumbled upon this thread. rembered i noticed this small "update available" notification in the HTTPie desktop app all day. 

Updated...

Export feature available :D 

Came back to check the age of the latest answer from @jkbrzt and realized it was released yesterday. What a coincidence. Love it.

**Comment by willianba** [comment]
i may be dumb but are you exporting in your own json spec?
i can't import the json file in any other app
